src/python/bcc/__init__.py

```diff
@@ -388,7 +388,6 @@ def attach_kprobe(self, event="", fn_name="", event_re="",
         return self
 
     def detach_kprobe(self, event):
-        assert isinstance(event, str), "event must be a string"
```

Comment:
I *think* the only case in which it's awkward is when using py2 with unicode_literals. In both py2 and 3 strings (e.g. "__kprobe_foo") are type `str`. Only when importing unicode_literals are they something else (a `unicode`). As long as we're fine with asserting in this case they can stay.

Alternatively, we can cast event to `str` in `attach_*`.